### ISSUE TITLE
PPO2: Non-deterministic behavior when setting --num_timesteps

### ISSUE THREAD

**Issue by brett-daley** [openai/baselines#878]
When I change the `--num_timesteps` argument for PPO2, even after setting the random seed and with all other hyperparameters the same, I get different performances after a few updates. **Although each individual run is entirely deterministic, for some reason changing the length of training affects the reproducibility of the results.** I'm on commit fa37beb.

For example, running PPO2 for 4 updates:
`python -m baselines.run --env=Hopper-v2 --seed=0 --num_env=1 --nsteps=2048 --num_timesteps=8192`

I get these episode rewards:
```
| eprewmean          | 16.5         |
| eprewmean          | 21.1         |
| eprewmean          | 23.1         |
| eprewmean          | 31.4         |
```

And then running it for 5 updates:
`python -m baselines.run --env=Hopper-v2 --seed=0 --num_env=1 --nsteps=2048 --num_timesteps=10240`

I get these:
```
| eprewmean          | 16.5         |
| eprewmean          | 21.1         |
| eprewmean          | 24.1         |
| eprewmean          | 32.9         |
| eprewmean          | 54.6         |
```

Notice that the performances have diverged at the 3rd update, despite the identical hyperparameters and seed. I believe this is a bug, but I cannot figure out where it is coming from.

**Comment by brett-daley** [comment]
I realized this behavior is due to the annealed learning rate, not a bug.
